File: php/hamming/hamming.php
<?php

function distance(string $strandA, string $strandB): int
{
    if (strlen($strandA) !== strlen($strandB)) {
        throw new InvalidArgumentException('DNA strands must be of equal length.');
    }
    $strandA = str_split($strandA);
    $strandB = str_split($strandB);
    return count(array_diff_assoc($strandA, $strandB));
}

File: python/hamming/hamming.py
def distance(strand_a, strand_b):
    if len(strand_a) != len(strand_b):
        raise ValueError("Left and right strands must be of equal length")

    distance = 0
    for a, b in zip(list(strand_a), list(strand_b)):
        if a != b:
            distance += 1

    return distance
